app/src/main/java/com/es/findsoccerplayers/ActivityMaps.java

```diff
@@ -211,24 +217,36 @@ public boolean onOptionsItemSelected(MenuItem item) {
     public void onMapReady(GoogleMap googleMap) {
         mMap = googleMap;
         mapReady = true;
-        enableMyLocation();
-        //If the user has a last known location the map will start from there
-        // if there is not a last known location the entire world map will appear and the user will find his location
-        // by pressing the button in the top right corner of the map.
 
-        setMapLongClick(mMap); // Add a marker in a long position click
-        setPoiClick(mMap);  //add a marker in a POI
-        setInfoWindowClick(mMap); // choose the selected position if the user clicks in the info window
-        onMapMoving(mMap); //if the camera moves because the user move it, stop tracking location. He's looking for a place.
 
-        if(initPosition != null){
+        if(locationAccess){
+            if(PositionClient.isGpsOFF(ActivityMaps.this)){
+                PositionClient.turnGPSon(ActivityMaps.this);
+            }
+        }
+
+        //If I have location access and GPS is on then show a small blue circle to identify my position
```

Comment:
mmmh, cosa cambia dall'if sopra? Stai ritestando le stesse condizioni

Reply:
Di quali if parli? quello in riga 222 e quello in riga 229?

Reply:
si